### DISCUSSION TITLE
How to set hostname for remote syslog?

### DISCUSSION THREAD

**Discussion by danmcclain** [gokrazy/gokrazy#232] [Q&A] [answered]
I'm poking around with homelabbing some stuff with a couple Gokrazy instances on my network and just got syslog-ng -> promtail -> loki setup. I noticed when pulling the logs that the gokrazy instances don't set the hostname. 

Is that a configuration issue or a missing feature at the moment? I figured the instance's configured `Hostname` would be reflected in the syslog hostname

**Comment by stapelberg** [comment, marked as answer]
gokrazy does use the hostname in its remote syslog messages, at least on my devices. See this tcpdump packet capture (expression `port 514`):

![2023-11-19-wireshark](https://github.com/gokrazy/gokrazy/assets/55506/5c8f4c45-192f-413b-9ed3-dbc3959d082b)

The [gokrazy syslog daemon](https://github.com/gokrazy/syslogd) separates log messages by host name, and I just now double-checked that my instance does indeed write to one logfile per hostname.

I wonder if you’re hitting a syslog format incompatibility in some part of your stack?

**Reply by danmcclain** [reply]
Turns out the default option for `keep-hostname` is `no` in syslog-ng. Thanks for helping me pinpoint!
